### ISSUE TITLE
debuginfo: Are underscores in tuple struct fields (`__0`, `__1` etc) required?

### ISSUE THREAD

**Issue by petrochenkov** [rust-lang/rust#36763]
Sorry for a silly question, but do they have any hidden debuginfo-specific meaning or it's just a style?

The code can be streamlined a bit if debuginfo uses fields simply by their names (`0`, `1`, `a`, `b`) regardless of their origin, like many other places in the compiler do.

cc @michaelwoerister

**Comment by Aatch** [comment]
A look at the DWARF spec doesn't seem to indicate that the names have to be in any format, so they could just be `0`, `1`, `2`, etc. That doesn't mean LLVM or MSVC will accept the names though.

**Comment by michaelwoerister** [comment]
If I remember correctly, GDB's parser didn't handle numbers as field names at the time. A newer Rust-aware version will probably not have that restriction (though I have not tried it).

**Comment by petrochenkov** [comment]
Ok, I experimented a bit.
Numeric field names are displayed correctly in gdb >= 7.9 both on Windows (mingw-w64) and Linux.
Pretty-printers need to be adjusted though.
Expression parsing doesn't work, i.e. `print s.0` is an error.
I wasn't able to get anything useful from lldb neither on Windows, nor on Linux, it doesn't seem to work with Rust programs.

Closing the issue, my curiosity is satisfied and there are no strong reasons to change debuginfo.
